fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java

```diff
@@ -0,0 +1,700 @@
+ */
+public class SparkLoadJob extends BulkLoadJob {
+    private static final Logger LOG = LogManager.getLogger(SparkLoadJob.class);
+
+    // for global dict
+    public static final String BITMAP_DATA_PROPERTY = "bitmap_data";
+
+    // --- members below need persist ---
+    // create from resourceDesc when job created
+    private SparkResource sparkResource;
+    // members below updated when job state changed to etl
+    private long etlStartTimestamp = -1;
+    // for spark yarn
+    private String appId = "";
+    // spark job outputPath
+    private String etlOutputPath = "";
+    // members below updated when job state changed to loading
+    // { tableId.partitionId.indexId.bucket.schemaHash -> (etlFilePath, etlFileSize) }
+    private Map<String, Pair<String, Long>> tabletMetaToFileInfo = Maps.newHashMap();
+
+    // --- members below not persist ---
+    // temporary use
+    // one SparkLoadJob has only one table to load
+    // hivedb.table for global dict
+    private String hiveTableName = "";
+    private ResourceDesc resourceDesc;
+    // for spark standalone
+    private SparkAppHandle sparkAppHandle;
+    // for straggler wait long time to commit transaction
+    private long quorumFinishTimestamp = -1;
+    // below for push task
+    private Map<Long, Set<Long>> tableToLoadPartitions = Maps.newHashMap();
+    private Map<Long, PushBrokerScannerParams> indexToPushBrokerReaderParams = Maps.newHashMap();
+    private Map<Long, Integer> indexToSchemaHash = Maps.newHashMap();
+    private Map<Long, Map<Long, PushTask>> tabletToSentReplicaPushTask = Maps.newHashMap();
+    private Set<Long> finishedReplicas = Sets.newHashSet();
+    private Set<Long> quorumTablets = Sets.newHashSet();
+    private Set<Long> fullTablets = Sets.newHashSet();
+
+    private static class PushBrokerScannerParams {
+        TBrokerScanRange tBrokerScanRange;
+        TDescriptorTable tDescriptorTable;
+
+        public void init(List<Column> columns, BrokerDesc brokerDesc) throws UserException {
+            Analyzer analyzer = new Analyzer(null, null);
+            // Generate tuple descriptor
+            DescriptorTable descTable = analyzer.getDescTbl();
+            TupleDescriptor destTupleDesc = descTable.createTupleDescriptor();
+            // use index schema to fill the descriptor table
+            for (Column column : columns) {
+                SlotDescriptor destSlotDesc = descTable.addSlotDescriptor(destTupleDesc);
+                destSlotDesc.setIsMaterialized(true);
+                destSlotDesc.setColumn(column);
+                if (column.isAllowNull()) {
+                    destSlotDesc.setIsNullable(true);
+                } else {
+                    destSlotDesc.setIsNullable(false);
+                }
+            }
+            // Push broker scan node
+            PushBrokerScanNode scanNode = new PushBrokerScanNode(destTupleDesc);
```

Comment:
Why is this ScanNode needed?
Seems that this scanNode will be destruct after this function.